src/main/java/com/amazon/ion/impl/IonTypeID.java

```diff
@@ -116,6 +117,10 @@ final class IonTypeID {
         // This is used as a dummy ID when a struct switches to using FlexSym field names in the middle. The key
         // here is that the type is STRUCT and the isInlineable flag is true.
         STRUCT_WITH_FLEX_SYMS_ID = TYPE_IDS_1_1[VARIABLE_LENGTH_STRUCT_WITH_FLEX_SYMS & 0xFF];
+
+        // This is used as a dummy ID when a delimited container reaches its end. The key here is that the type ID's
+        // lower nibble is OpCodes.DELIMITED_END_MARKER.
+        DELIMITED_END_ID = TYPE_IDS_1_1[DELIMITED_END_MARKER & 0xFF];
```

Comment:
Unfortunately it's not redundant, because DELIMITED_END_MARKER is stored as a `byte`... So, when it's widened to an `int` to serve as an array index, its most-significant bit is extended to produce a negative value unless you use the one-byte mask.